uma/exceptions.py

```diff
@@ -2,32 +2,69 @@
 
 import json
 from typing import List
+from uma.generated.errors import ErrorCode
 
 
-class UnsupportedVersionException(Exception):
+class UmaException(Exception):
+    def __init__(
+        self, reason: str = "", error_code: ErrorCode = ErrorCode.INTERNAL_ERROR
```

Comment:
Do we definitely want defaults for these params or do we want to force ourselves to remember to set it explicitly?